### PR TITLE
chore: overhead hp bar

### PR DESCRIPTION
**Pull request by Arufonsu**
* Simplified code, tested and covered all scenarios.

### CODE REVIEW

**Review by WeylonSantana — APPROVED** [review]

**Comment by Arufonsu** [comment]
For the record: the following was untested until now 🧪 


https://user-images.githubusercontent.com/17498701/214177794-c7473780-2074-4add-985a-7edd53dc50ae.mp4

File: Intersect.Client/Entities/Entity.cs [hunk 1]
```diff
@@ -1411,29 +1411,21 @@ protected virtual bool ShouldDrawHpBar
...
             {
-                if (!ShouldNotDrawHpBar || IsHovered)
+                if (ShouldNotDrawHpBar)
                 {
-                    return true;
+                    return false;
                 }
 
-                return GetType() == typeof(Entity) && Globals.Database.NpcOverheadHpBar;
+                return (GetType() == typeof(Entity) && Globals.Database.NpcOverheadHpBar) ||
```
Comment (lodicolo) [inline Intersect.Client/Entities/Entity.cs:1419]: Check IsHovered first, then shield, then HP, and only then should you check the entity type.

This should also be broken up into multiple if statements so it isn't such a long line.

File: Intersect.Client/Entities/Player.cs [hunk 2]
```diff
@@ -2383,32 +2383,13 @@ protected override bool ShouldDrawHpBar
...
-                }
-
-                return Globals.Database.PlayerOverheadHpBar;
+                var me = Globals.Me;
+                bool userPreference = (Globals.Database.MyOverheadHpBar && Id == me.Id) ||
+                                      (Globals.Database.PlayerOverheadHpBar && Id != me.Id) ||
+                                      (Globals.Database.PartyMemberOverheadHpBar && me.IsInMyParty(this)) ||
+                                      (Globals.Database.FriendOverheadHpBar && me.IsFriend(this)) ||
+                                      (Globals.Database.GuildMemberOverheadHpBar && me.IsGuildMate(this));
+                return userPreference || IsHovered;
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:2392]: Revert this. This is not more readable, and forces expensive operations to happen before IsHovered which is a much less costly check.

The original structure and order was very intentional.

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect.Client/Entities/Player.cs [hunk 3]
```diff
@@ -2388,27 +2388,31 @@ protected override bool ShouldDrawHpBar
...
-                    return true;
+                    userPreference = true;
+                }
+                else if (Globals.Database.GuildMemberOverheadHpBar && me.IsGuildMate(this))
+                {
+                    userPreference = true;
                 }
 
-                return Globals.Database.PlayerOverheadHpBar;
+                return userPreference;
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:2415]: Get rid of this extra variable, and all of the `else` keywords.

Just return true (and return <the last if statement>)

**Review by lodicolo — CHANGES_REQUESTED** [review]

**Review by lodicolo — APPROVED** [review]

**Pull request merged**